scripts/release/stage-release.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+set -ex
+
+# Stages a release by putting everything that should be packaged and released
+# into the ./deploy folder. This script should be run from the root of the
+# material2 repo.
+
+# Make sure you are not running `ng serve` or `ng build --watch` when running this.
+
+
+# Clear dist/ and deploy/ so that we guarantee there are no stale artifacts.
+rm -rf ./dist
+rm -rf ./deploy
+
+# For packaging for npm only, replace the TypeScript module format to commonjs
+sed -i.backup 's|"module": ".+"|"module": "commonjs"|g' ./src/tsconfig.json
+
+# Perform a build with the modified tsconfig.json.
+ng build
+
+# Return tsconfig.json to its original state.
+mv -f ./src/tsconfig.json.backup ./src/tsconfig.json
+
+# Inline the css and html into the component ts files.
+./node_modules/gulp/bin/gulp.js inline-resources
+
+# deploy/ serves as a working directory to stage the release.
+mkdir deploy
+
+# Copy all components/ to deploy/ and replace "../core" with just "core" since each
+# component directory will now live as a sibling to core/
+cp -R ./dist/components/* ./deploy/
+find ./deploy -type f \( -name "*.js" -o -name "*.ts" \) -exec sed -i '' 's|\.\./core|core|g' {} \;
+
+# Copy the core/ directory directly into ./deploy
+cp -R ./dist/core/ ./deploy/core/
+
+# Create tarball packages for local testing (not for publishing / distribution)
+# (if installing these packages doesn't work, ensure your npm version is 3.5.0 or greater)
+find ./deploy/* -type d -maxdepth 0 | xargs npm pack
+mkdir ./deploy/packages/
+mv *.tgz ./deploy/packages/
```

Comment:
can you also document instructions for how to test the build locally? it will be useful for people not familiar with npm link (or npm pack if you stick to that)